### PR TITLE
Read the admin password from a file

### PR DESCRIPTION
**Pull request by pnmadelaine**
Instead of using the digest of the admin password, Typhon now expects a path to a file containing the password.
This allows to use frameworks like agenix for deployment, and avoids letting digests that could be sensitive to rainbow attacks into public configurations.

### CODE REVIEW

File: typhon/src/main.rs [hunk 1]
```diff
@@ -45,9 +48,12 @@ async fn main() -> std::io::Result<()> {
     tracing::subscriber::set_global_default(tracing_subscriber::FmtSubscriber::new()).unwrap();
 
     let args = Args::parse();
-    std::env::set_var("HASHED_PASSWORD", args.hashed_password);
 
     // Initialization
+    let mut password_file = File::open(args.password)?;
+    let mut password = String::new();
+    password_file.read_to_string(&mut password)?;
+    std::env::set_var("PASSWORD", password);
```
Comment (W95Psp) [inline typhon/src/main.rs:56]: Oh, I never saw that: we are setting an env var for the current process which in turn is read by typhon-core?
We should do something different there!

But that's orthogonal to this PR. I filed issue #7.

**Review by W95Psp — APPROVED** [review]
Looks good to me!
Maybe we also want to keep a `--password` flag (exclusive with `--password-file`), wdyt?

File: typhon/src/main.rs [hunk 2]
```diff
@@ -45,9 +48,12 @@ async fn main() -> std::io::Result<()> {
     tracing::subscriber::set_global_default(tracing_subscriber::FmtSubscriber::new()).unwrap();
 
     let args = Args::parse();
-    std::env::set_var("HASHED_PASSWORD", args.hashed_password);
 
     // Initialization
+    let mut password_file = File::open(args.password)?;
+    let mut password = String::new();
+    password_file.read_to_string(&mut password)?;
```
Comment (W95Psp) [inline typhon/src/main.rs:55]:
```suggestion
    let password = std::fs::read_to_string("/etc/hosts")?;
```

**Comment by pnmadelaine** [comment]
> Looks good to me! Maybe we also want to keep a `--password` flag (exclusive with `--password-file`), wdyt?

If the option is not secure my opinion is that we should not expose it.
Also, I would keep this as simple as possible until we do proper user authentication.

**Pull request merged**